src/main/scala/viper/gobra/translator/encodings/combinators/FinalTypeEncoding.scala

```diff
@@ -28,20 +27,39 @@ class FinalTypeEncoding(te: TypeEncoding) extends TypeEncoding {
   }
 
   override def finalize(addMemberFn: vpr.Member => Unit): Unit = te.finalize(addMemberFn)
+
+  /* A result is not guaranteed */
+  override def varPrecondition(ctx: Context): in.Parameter.In ==> MemberWriter[vpr.Exp] = te.varPrecondition(ctx)
+  override def varPostcondition(ctx: Context): in.Parameter.Out ==> MemberWriter[vpr.Exp] = te.varPostcondition(ctx)
+
+  /* A result is guaranteed */
   override def typ(ctx: Context): in.Type ==> vpr.Type = te.typ(ctx) orElse expectedMatch("typ")
   override def variable(ctx: Context): in.BodyVar ==> vpr.LocalVarDecl = te.variable(ctx) orElse expectedMatch("variable")
   override def globalVar(ctx: Context): in.GlobalVar ==> CodeWriter[vpr.Exp] = te.globalVar(ctx) orElse expectedMatch("globalVar")
-  override def member(ctx: Context): in.Member ==> MemberWriter[Vector[vpr.Member]] = te.member(ctx)
-  override def precondition(ctx: Context): in.Parameter.In ==> MemberWriter[vpr.Exp] = te.precondition(ctx)
-  override def postcondition(ctx: Context): in.Parameter.Out ==> MemberWriter[vpr.Exp] = te.postcondition(ctx)
+  override def method(ctx: Context): in.Member ==> MemberWriter[vpr.Method] = te.method(ctx) orElse expectedMatch("method")
+  override def function(ctx: Context): in.Member ==> MemberWriter[vpr.Function] = te.function(ctx) orElse expectedMatch("function")
+  override def predicate(ctx: Context): in.Member ==> MemberWriter[vpr.Predicate] = te.predicate(ctx) orElse expectedMatch("predicate")
+  override def member(ctx: Context): in.Member ==> MemberWriter[Vector[vpr.Member]] = te.member(ctx) orElse expectedMatch("member")
```

Comment:
the member is the function used to encode the members of the input program and its results are added to the Viper program. The difference to the other three functions is that these can return something in cases where member has to return an emtpy vector to avoid duplicates. I give a full explanation in the documentation of the member method.